docs/src/main/docs/guides/07_openapi.adoc

```diff
@@ -0,0 +1,240 @@
+that describes your application's endpoints.
+
+== What you need
+
+|===
+|About 15 minutes
+|<<about/03_prerequisites.adoc,Helidon Prerequisites>>
+|a Helidon MP (or SE) application to update for OpenAPI 
+|===
+
+== OpenAPI for Helidon MP
+Helidon MP conforms to the link:{mp-openapi-spec}[MicroProfile OpenAPI specification],
+which was inspired by the link:{openapi-spec}[OpenAPI spec] itself.
+
+You can follow two general approaches with OpenAPI in your Helidon MP app:
+
+1. add MicroProfile OpenAPI annotations to the endpoints in your source code, allowing
+the Helidon MP runtime to discover those endpoints via CDI at app start-up; and
+2. create a static OpenAPI document describing your app's endpoints, for example
+using a separate tool such as Swagger, and packaging it with your app.
+
+You can use either or both in the same application. Helidon MP support for OpenAPI
+will combine these and other sources of endpoint information when it serves the
+OpenAPI document for your app. (See 
+<<Advanced OpenAPI support in Helidon SE and MP,the advanced section>> below.)
+
+You also need to update your `pom.xml`.
+
+The next sections show you how to make these changes. Helidon also includes a 
+link:{helidon-mp-openapi-example}[complete example maven project] showing
+Helidon MP OpenAPI support.
+
+=== Annotating endpoints
+The added OpenAPI-related annotations allow the system to build the OpenAPI document 
+describing your endpoints. 
+
+Here is one of the endpoints from the example mentioned earlier:
+
+[source,java]
+----
+@GET
+@Operation(summary = "Returns a generic greeting", // <1>
+        description = "Greets the user generically")
+@APIResponse(description = "Simple JSON containing the greeting", // <2>
+        content = @Content(mediaType = "application/json",
+                           schema = @Schema(implementation = GreetingMessage.class)))
+@Produces(MediaType.APPLICATION_JSON)
+public JsonObject getDefaultMessage() {...}
+----
+<1> `@Operation` gives information about this endpoint.
+<2> `@APIResponse` describes the HTTP response and declares its media type and contents.
+
+You can also define any request parameters the endpoint expects, although this
+endpoint uses none.
+
+This guide shows only a few annotations for illustration. The 
+link:{helidon-mp-openapi-example}[Helidon MP OpenAPI example] illustrates more, 
+and the link:{mp-openapi-spec}[MicroProfile OpenAPI spec] describes them all.
+
+=== Updating your `pom.xml`
+==== Building the Jandex index
+A Jandex index stores information about the classes and methods in your app and
+what annotations they have. It allows CDI to process annotations faster during your 
+application's start-up.
+
+Add the link:{jandex-plugin-doc}[Jandex maven plug-in] to the `<build><plugins>` 
+section of your `pom.xml`:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.jboss.jandex</groupId>
+    <artifactId>jandex-maven-plugin</artifactId>
```

Comment:
Probably should have a `<version>` tag here, or include a `<pluginManagement>` stanza with the version specified.  I believe it must be 1.0.6 or later, yes? (Can't remember offhand.)

Reply:
Yes. I'll do that. Our example from which I drew all the code benefits from the inherited `<pluginManagement>` but a user won't be inheriting from our poms. Luckily it's pretty easy to pass the parameterized version we use in our build into the ADOC processing so when we build the site for a version of Helidon we capture the version we've used ourselves.